### PR TITLE
Add WeMM embedding model.

### LINKED ISSUE #5302: Add model: WeMM-Embedding
### Model link on Hugging Face

https://huggingface.co/collections/tencent/wemm-embedding

### Arxiv link

_No response_

### Description of the model

PR with integration with SentenceTransformers https://huggingface.co/tencent/WeMM-Embedding-2B/discussions/1

### PR DESCRIPTION
**Pull request by deep9539**
Close https://github.com/embeddings-benchmark/mteb/issues/5302

- [x] I have filled out the ModelMeta object to the extent possible
- [x] I have ensured that my model can be loaded using
  - [x] `mteb.get_model(model_name, revision)` and
  - [x] `mteb.get_model_meta(model_name, revision)`
- [x] I have tested the implementation works on a representative set of tasks.
- [x] The model is public, i.e., is available either as an API or the weights are publicly available to download
- [?] I reproduced results from the original paper (if applicable) on at least one benchmark, and I am including the results in the PR description.

## Verification

(Running VATEXT2VRetrieval, MSRVTTT2V & YouCook2T2VRetrieval, taking too long to run due to `batch_size: 1`.)

To verify the implementation of **WeMM-Embedding-2B**, I ran evaluations on the standard image-to-text retrieval tasks: `Flickr30kI2TRetrieval` and `MSCOCOI2TRetrieval`.

While the paper lists "Recall@1" scores of **95.6%** (Flickr30k) and **78.8%** (MSCOCO), our evaluated `recall_at_1` scores came out to **19.5%** and **15.9%** respectively. Comparing our scores against other top multimodal models registered in MTEB confirmed that this 10%–20% range is the standard expected baseline for `recall_at_1` on these benchmarks.

To verify the implementation of **WeMM-Embedding-2B**, I ran evaluations on the standard image-to-text retrieval tasks: `Flickr30kI2TRetrieval` and `MSCOCOI2TRetrieval`.

While the paper lists "Recall@1" scores of **95.6%** (Flickr30k) and **78.8%** (MSCOCO), our evaluated `recall_at_1` scores came out to **19.5%** and **15.9%** respectively. Comparing our scores against other top multimodal models registered in MTEB confirmed that this 10%–20% range is the standard expected baseline for `recall_at_1` on these benchmarks.

🔍 **Resolving the Discrepancy (IR Recall vs. Hit Rate)**
This is a standard nomenclatural mismatch in multimodal literature:
1. **MTEB's `recall_at_1`**: Follows strict Information Retrieval (IR) standards, normalizing the score by dividing by the total count of relevant documents available in the corpus for that query.
2. **The Paper's "Recall@1"**: In multimodal retrieval papers (including WeMM), "Recall@K" is almost always mathematically defined as a Hit Rate—the percentage of queries where at least one ground-truth target is retrieved in the top K.

📈 **Verification via `hit_at_1`**
When evaluating WeMM under MTEB's `hit_at_1` (which corresponds exactly to the paper's definition of "Recall@1"), the scores align perfectly with the reported numbers:

| Dataset | Metric in Paper | Measured `recall_at_1` | Measured `hit_at_1` (Expected "Recall@1") |
| :--- | :--- | :--- | :--- |
| Flickr30kI2TRetrieval | 95.6 | 19.5 | 97.3 (Close match) |
| MSCOCOI2TRetrieval | 78.8 | 15.9 | 79.4 (Close match) |

### CODE REVIEW

File: mteb/models/model_implementations/wemm_models.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, cast
+
+import numpy as np
+import torch
+from torchvision.transforms.functional import to_pil_image
```
Comment (Samoed) [inline mteb/models/model_implementations/wemm_models.py]: This shoud be inside model class

File: mteb/models/model_implementations/wemm_models.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                    ]
+                    if isinstance(v, torch.Tensor)
+                    else v
+                    for v in videos
+                ]
+
+            # C. Build unified list of inputs (string for text, dict for multimodal)
+            batch_size = (
+                len(texts) if texts else (len(images) if images else len(videos))
+            )
```
Comment (Samoed) [inline mteb/models/model_implementations/wemm_models.py]: You can get `batch_size` that passed to `encode` method

File: mteb/models/model_implementations/wemm_models.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                    if videos and videos[i] is not None:
+                        sample["video"] = videos[i]
+                    if text_content:
+                        sample["text"] = text_content
+                    batched_input.append(sample)
+                else:
+                    batched_input.append(text_content)
+
+            # Call SentenceTransformer's encode with prompt=None to bypass system-role wrapping
+            embeddings = self.model.encode(batched_input, prompt=None, **kwargs)
```
Comment (Samoed) [inline mteb/models/model_implementations/wemm_models.py]: Why `prompt=None`?

File: mteb/models/model_implementations/wemm_models.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+        super().__init__(
+            model_name=model_name,
+            revision=revision,
+            device=device,
+            **kwargs,
+        )
+        self.fps = fps
+        self.max_frames = max_frames
+        self.num_frames = num_frames
+        self.target_sampling_rate = target_sampling_rate
```
Comment (Samoed) [inline mteb/models/model_implementations/wemm_models.py]: This will be set in base class
```suggestion
```

File: mteb/models/model_implementations/wemm_models.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+                target_sampling_rate=self.target_sampling_rate,
+                fps=self.fps,
+                max_frames=self.max_frames,
+                num_frames=self.num_frames,
+            )
+
+        instruction = self.get_task_instruction(task_metadata, prompt_type)
+        all_embeddings = []
+
+        # B. Process batches from MTEB's DataLoader
```
Comment (Samoed) [inline mteb/models/model_implementations/wemm_models.py]: I don't think we need such comments
```suggestion
```

File: mteb/models/model_implementations/wemm_models.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+                        # Convert HWC tensor format to CHW before calling to_pil_image
+                        to_pil_image(
+                            f.permute(2, 0, 1)
+                            if f.ndim == 3 and f.shape[-1] == 3
+                            else f
+                        )
+                        for f in v.cpu()
+                    ]
+                    if isinstance(v, torch.Tensor)
+                    else v
```
Comment (Samoed) [inline mteb/models/model_implementations/wemm_models.py]:
```suggestion
```

**Review by Samoed — COMMENTED** [review]
Can you measure ViDoRe results?

@JUNJIE99 @Raykoooo would be great to get your reviews for your model implementation

File: mteb/models/model_implementations/wemm_models.py [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+    name="tencent/WeMM-Embedding-9B",
+    model_type=["dense"],
+    modalities=["text", "image", "video"],
+    languages=["zho-Hans", "eng-Latn"],
+    open_weights=True,
+    revision="59f673e3d53c4f71bb364adf344536dc1c2bc95e",
+    release_date="2026-08-25",
+    n_parameters=9_000_000_000,
+    n_embedding_parameters=1_016_127_488,
+    memory_usage_mb=None,
```
Comment (ayush1298) [inline mteb/models/model_implementations/wemm_models.py]: can you fill these for all models

File: mteb/models/model_implementations/wemm_models.py (hunk 2 above)
Comment (deep9539) [reply mteb/models/model_implementations/wemm_models.py]: IIUC, since we are iterating `for i in range(batch_size):`, if the batch has only 20 rows, but `kwargs[batch_size]` is 32, then the code will fail.

File: mteb/models/model_implementations/wemm_models.py (hunk 3 above)
Comment (deep9539) [reply mteb/models/model_implementations/wemm_models.py]: it's part of batched_input, dropping this.

File: mteb/models/model_implementations/wemm_models.py (hunk 7 above)
Comment (deep9539) [reply mteb/models/model_implementations/wemm_models.py]: I think you mean memory_usage_mb, other fields were already there.

**Comment by deep9539** [comment]
> Can you measure ViDoRe results?

@Samoed Any particular reason to add ViDoRe results. I've added Flickr30k and MSCoco which do cover Image & text modality similar to ViDoRe?

**Comment by Samoed** [comment]
> Any particular reason to add ViDoRe results

I'm not sure that we have same data for flickr and mscoco tasks with mmeb

**Comment by deep9539** [comment]
> > Any particular reason to add ViDoRe results
> 
> I'm not sure that we have same data for flickr and mscoco tasks with mmeb

@Samoed Unfortunately the cluster is too busy, so not able to test this on ViDoRe. Do you want to wait for it to merge, or allow me to take a look at the numbers we come up with during the final run? 
Give the straight-forward implementation and number, I'm comfortable to submit as it is.

**Comment by Samoed** [comment]
I think better to wait a bit

**Comment by deep9539** [comment]
| Task Name | Average ndcg_at_10 |
|---|---|
| Vidore2BioMedicalLecturesRetrieval | 0.69055 |
| Vidore2ESGReportsHLRetrieval | 0.68881 |
| Vidore2ESGReportsRetrieval | 0.63884 |
| Vidore2EconomicsReportsRetrieval | 0.58509 |
| **Overall Average** | **0.65082** |

Paper reported average of 66.3 for WeMM-Embed-9B. Ours is very close around 65.1

File: mteb/models/model_implementations/wemm_models.py (hunk 2 above)
Comment (deep9539) [reply mteb/models/model_implementations/wemm_models.py]: (Resolving)

**Review by Samoed — APPROVED** [review]
